plugins/ocm-backend/config.d.ts

```diff
@@ -29,14 +29,24 @@ interface HubClusterConfig {
   caData?: string;
 }
 
+interface Owner {
+  owner:
+    | string
+    | {
+        kind?: string;
+        namespace?: string;
+        name: string;
+      };
+}
+
 export interface Config {
   catalog?: {
     providers?: {
       ocm?: {
         /**
          * Key is reflected as provider ID. Defines and claims plugin instance ownership of entities
          */
-        [key: string]: KubernetesPluginRef | HubClusterConfig;
+        [key: string]: (KubernetesPluginRef | HubClusterConfig) | Owner;
```

Comment:
This is not correct, following passes schema validation:


```
catalog:
  providers:
    ocm:
      foo:
        kind: User
        name: test
```

Reply:
Thats true, I changed so that the schema looks like this now:
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "Config": {
      "additionalProperties": false,
      "properties": {
        "catalog": {
          "additionalProperties": false,
          "properties": {
            "providers": {
              "additionalProperties": false,
              "properties": {
                "ocm": {
                  "additionalProperties": {
                    "anyOf": [
                      {
                        "additionalProperties": false,
                        "properties": {
                          "kubernetesPluginRef": {
                            "description": "Match the cluster name in kubernetes plugin config",
                            "type": "string"
                          },
                          "owner": {
                            "anyOf": [
                              {
                                "type": "string"
                              },
                              {
                                "additionalProperties": false,
                                "properties": {
                                  "kind": {
                                    "enum": [
                                      "user",
                                      "group"
                                    ],
                                    "type": "string"
                                  },
                                  "name": {
                                    "type": "string"
                                  },
                                  "namespace": {
                                    "type": "string"
                                  }
                                },
                                "required": [
                                  "name"
                                ],
                                "type": "object"
                              }
                            ]
                          }
                        },
                        "required": [
                          "kubernetesPluginRef"
                        ],
                        "type": "object"
                      },
                      {
                        "additionalProperties": false,
                        "properties": {
                          "caData": {
                            "description": "Base64-encoded CA bundle in PEM format used for verifying the TLS cert presented by the API server",
                            "type": "string"
                          },
                          "name": {
                            "description": "Name that the hub cluster will assume in Backstage Catalog (in OCM this is always local-cluster which can be confusing)",
                            "type": "string"
                          },
                          "owner": {
                            "anyOf": [
                              {
                                "type": "string"
                              },
                              {
                                "additionalProperties": false,
                                "properties": {
                                  "kind": {
                                    "enum": [
                                      "user",
                                      "group"
                                    ],
                                    "type": "string"
                                  },
                                  "name": {
                                    "type": "string"
                                  },
                                  "namespace": {
                                    "type": "string"
                                  }
                                },
                                "required": [
                                  "name"
                                ],
                                "type": "object"
                              }
                            ]
                          },
                          "serviceAccountToken": {
                            "description": "Service Account Token which is used for querying data from the hub",
                            "type": "string"
                          },
                          "skipTLSVerify": {
                            "description": "Skip TLS certificate verification presented by the API server, defaults to false",
                            "type": "boolean"
                          },
                          "url": {
                            "description": "Url of the hub cluster API endpoint",
                            "type": "string"
                          }
                        },
                        "required": [
                          "name",
                          "url"
                        ],
                        "type": "object"
                      }
                    ],
                    "description": "Key is reflected as provider ID. Defines and claims plugin instance ownership of entities"
                  },
                  "type": "object"
                }
              },
              "type": "object"
            }
          },
          "type": "object"
        }
      },
      "type": "object"
    },
    "HubClusterConfig": {
      "additionalProperties": false,
      "properties": {
        "caData": {
          "description": "Base64-encoded CA bundle in PEM format used for verifying the TLS cert presented by the API server",
          "type": "string"
        },
        "name": {
          "description": "Name that the hub cluster will assume in Backstage Catalog (in OCM this is always local-cluster which can be confusing)",
          "type": "string"
        },
        "serviceAccountToken": {
          "description": "Service Account Token which is used for querying data from the hub",
          "type": "string"
        },
        "skipTLSVerify": {
          "description": "Skip TLS certificate verification presented by the API server, defaults to false",
          "type": "boolean"
        },
        "url": {
          "description": "Url of the hub cluster API endpoint",
          "type": "string"
        }
      },
      "required": [
        "url",
        "name"
      ],
      "type": "object"
    },
    "KubernetesPluginRef": {
      "additionalProperties": false,
      "properties": {
        "kubernetesPluginRef": {
          "description": "Match the cluster name in kubernetes plugin config",
          "type": "string"
        }
      },
      "required": [
        "kubernetesPluginRef"
      ],
      "type": "object"
    },
    "Owner": {
      "additionalProperties": false,
      "properties": {
        "owner": {
          "anyOf": [
            {
              "type": "string"
            },
            {
              "additionalProperties": false,
              "properties": {
                "kind": {
                  "enum": [
                    "user",
                    "group"
                  ],
                  "type": "string"
                },
                "name": {
                  "type": "string"
                },
                "namespace": {
                  "type": "string"
                }
              },
              "required": [
                "name"
              ],
              "type": "object"
            }
          ]
        }
      },
      "type": "object"
    }
  }
}

```

where the owner field is optional but can't be as the only key in a hub config.